### DISCUSSION TITLE
Response cannot be a simple type like int

### DISCUSSION THREAD

**Discussion by cropr** [luolingchun/flask-openapi3#51] [Q&A] [answered]
Environment:

- Python version: 3.10
- Operating system: tested it on Windows 10 and Ubuntu 22.04
- Flask version: 2.0.3
- flask-openapi3 version: 2.0.0

I cannot create an endpoint that is returning an int (or any other native type)
```
@app.get("/path", responses={"200": int})
def apicall():
    return 1

```

gives error TypeError: <class 'int'> is invalid `pydantic.BaseModel`.

I could leave the responses out, but this makes the generated doc incomplete

**Comment by dvaerum** [comment, marked as answer]
Try to do something like this
```python
class ResponseWithInt(BaseModel):
    __root__: int
```
other examples: https://github.com/luolingchun/flask-openapi3/blob/master/tests/test_restapi.py#L69-L70

**Comment by luolingchun** [comment]
> Try to do something like this
> 
> ```python
> class ResponseWithInt(BaseModel):
>     __root__: int
> ```

I agree, but the `Media type` is `application/json`.

![image](https://user-images.githubusercontent.com/22740403/176572224-6996df00-e015-478e-b7b0-e206c0fb1677.png)


So is it better to use `extra_responses`?

```python
# Customized media type `text/plain `
@app.get("/path", {"200": {"content": {"text/plain": {"schema": {"type": "integer"}}}}})
def apicall():
    return 1
```

![image](https://user-images.githubusercontent.com/22740403/176572391-1ac75631-6bf3-46f2-968a-d549571aedc6.png)

**Comment by cropr** [comment]
> Try to do something like this
> 
> ```python
> class ResponseWithInt(BaseModel):
>     __root__: int
> ```
> 
> other examples: https://github.com/luolingchun/flask-openapi3/blob/master/tests/test_restapi.py#L69-L70

That does indeed solve my question
